Mistcloak v2

SoaringEaqle opened the pull request:
## Changes proposed in this pull request:
Fixes mistcloak render by making it a Curios item to go in the head slot.

SoaringEaqle committed:
Mistcloak and Shardplate

SoaringEaqle committed:
rename

leafreynolds committed:
[surgebinding] fix issue where shardplate capability isn't synced to client correctly

SoaringEaqle committed:
Adds Shardplate
Shardplate buffs now handled by shardplate item.
Shardplate now has a charge. this can be refilled by placing broams and the shardplate in a crafting grid, or it can siphon off a player's stormlight.

Signed-off-by: SoaringEaqle <158627775+SoaringEaqle@users.noreply.github.com>

SoaringEaqle commented:
There's some stuff that still needs done, but it's mostly to do with how radiant shardplate works and interacts with radiants, so it's probably best to add it after radiants get added. Instead of working around a system that barely exists yet.

SoaringEaqle committed:
New Shardplate and blade stuff

Data for Shardplate and Blades -- including order, state of living or deadeye, who their bonded to, and how long they've been trying to bond to a player -- are now controlled by capabilities.

Suits of shardplate for each order have been removed.

The color of shardplate is now determined by the capability and its data.

SoaringEaqle commented:
Needed a break from Investiture System, so mostly redid this. Hopefully this redesign makes it easier to influence data, and manipulate deadeye states, as well as control bonding easier.

SoaringEaqle committed:
Merge branch 'develop' into mistcloak_V2

SoaringEaqle committed:
Merge branch 'develop' into mistcloak_V2

SoaringEaqle committed:
Mistcloak and Shard changes

Bug fixes for shard blades and plate
Color changes for blade and plate
new mistcloak texture.
Mistcloak now functions as a weak armor, and reduces player gravity allowing for better steeljumping.

SoaringEaqle committed:
Merge branch 'mistcloak_V2' of https://github.com/SoaringEaqle/cosmere into mistcloak_V2

SoaringEaqle committed:
New mistcloak texture

Signed-off-by: SoaringEaqle <158627775+SoaringEaqle@users.noreply.github.com>

SoaringEaqle committed:
mistcloak recipe

Signed-off-by: SoaringEaqle <158627775+SoaringEaqle@users.noreply.github.com>

Gerbagel commented:
Don't see any issues with the Mistcloak. 

However, when I go to change out one piece of Shardplate for another, the second one becomes the same type as the first. This happens regardless of Dead or living plate. 

`SurgebindingItemModelsGen.java` is also missing a couple imports for `GemstoneItem` and `ShardplateCurioItem`. 

I also get `java.lang.IllegalArgumentException: Texture surgebinding:item/gemstone_chip does not exist in any known resource pack` when I try to `runData`, which I was doing to see if the lack of a Curios slot name for Shardplate was just because I hadn't generated.

leafreynolds committed:
Merge branch 'develop' into mistcloak_V2

SoaringEaqle commented:
I did notice that, however, I do not know how to fix it. My assumption is that it is a quirk with curios. I think the best way to bypass it would be to store the data somewhere when you try to equip, and come back a few ticks later and reattach the data, but I really have no clue what that would look like.

SoaringEaqle committed:
Fixes issues

SoaringEaqle committed:
Revert "Fixes issues"

This reverts commit 9b813d483fef4e25cf3fec616093cb7240a885b4.

leafreynolds commented on src/surgebinding/java/leaf/cosmere/surgebinding/common/items/HonorbladeItem.java:
```diff
@@ -63,10 +75,49 @@ public Multimap<Attribute, AttributeModifier> getAttributeModifiers(EquipmentSlo
+	public void inventoryTick(ItemStack pStack, Level pLevel, Entity pEntity, int pItemSlot, boolean pIsSelected)
+	{
+		super.inventoryTick(pStack, pLevel, pEntity, pItemSlot, pIsSelected);
+		getShardData(pStack).setOrder(getOrder(pStack));
```
ideally we don't want this getting called every single tick

leafreynolds replied:
ideally we don't want this getting called every single tick

leafreynolds commented:
In future, we also don't want multiple features packed into a single pull request. This one is called mistcloak, but is mostly full of surgebinding stuff.

SoaringEaqle commented:
Sorry about that. I just needed a bunch of stuff I made for the mistcloak for the Shardplate, and I didn't want to go through the effort of duplicating the branch and removing the mistcloak-specific code.

SoaringEaqle committed:
Reapply "Fixes issues"

This reverts commit 3a9db2359742150a4e9cfc7015405ebfebc9033f.

SoaringEaqle commented:
> `SurgebindingItemModelsGen.java` is also missing a couple imports for `GemstoneItem` and `ShardplateCurioItem`.
> 
> I also get `java.lang.IllegalArgumentException: Texture surgebinding:item/gemstone_chip does not exist in any known resource pack` when I try to `runData`, which I was doing to see if the lack of a Curios slot name for Shardplate was just because I hadn't generated.

These two issues should be resolved. It was some issue where the files half updated, but not completely.

SoaringEaqle committed:
[Allomany] separates mistcloak into its own branch and PR

leafreynolds committed:
Remove all .dstore files from subdirectories

leafreynolds committed:
ignore .dsstore files

SoaringEaqle committed:
Merge branch 'develop' into mistcloak_V2

leafreynolds commented on src/surgebinding/java/leaf/cosmere/surgebinding/common/entity/spren/Honorspren.java:
```diff
@@ -35,7 +35,7 @@ public boolean canBeHitByProjectile()
 	public boolean canBeSeenByAnyone()
 	{
-		//TODO: make this only visible to the player it is bonded to?
+		//TODO: make this only visible to the player it is bond to?
```
"is bond to" isn't good english 😅 also please try not to commit changes like these where you're only changing a comment

leafreynolds commented on src/surgebinding/java/leaf/cosmere/surgebinding/common/items/ShardbladeItem.java:
```diff
@@ -54,9 +74,100 @@ public boolean isFoil(ItemStack itemStack)
+			}
+			else if(data.bondTicks() >= bondTime())
+			{
+				bond(pStack, player);
```
I'm not sure I understand this mechanic right for shardplate. You don't really bond plate like you would a radiant spren. You kinda have the plate spren, or you don't

leafreynolds commented on src/surgebinding/java/leaf/cosmere/surgebinding/common/items/ShardplateCurioItem.java:
```diff
@@ -0,0 +1,396 @@
+
+		// Copy the capability data if present
+		prevStack.getCapability(ShardData.SHARD_DATA).ifPresent(fromCap -> {
+			stack.getCapability(ShardData.SHARD_DATA).ifPresent(toCap -> {
```
Why are we copying the capability from old stack to new stack? does it get lost somehow?

Gerbagel replied:
Perhaps this has something to do with my previous comment? 

https://github.com/leafreynolds/cosmere/pull/159#issuecomment-3535051024

leafreynolds replied:
Oh probably. These comments on my review were from a while ago, I just didn’t push submit apparently. 

I’ll take a proper look tomorrow.

SoaringEaqle replied:
Curios creates a new itemstack when going into and out of a slot. The capability doesn't carry, so I create a new one and force the data over. I suppose I could try to figure out a way to make the capability transfer over instead, but this is working, so I'd rather not touch it.

leafreynolds commented on src/surgebinding/java/leaf/cosmere/surgebinding/common/items/ShardplateCurioItem.java:
```diff
@@ -0,0 +1,396 @@
+		stack.getCapability(ShardData.SHARD_DATA).ifPresent(fromCap -> {
+			newStack.getCapability(ShardData.SHARD_DATA).ifPresent(toCap -> {
+				CompoundTag nbt = fromCap.serializeNBT();
+				toCap.deserializeNBT(nbt);
```
same here, a bit confused.

leafreynolds commented on src/surgebinding/java/leaf/cosmere/surgebinding/common/items/ShardplateCurioItem.java:
```diff
@@ -0,0 +1,396 @@
+
+
+	@Override
+	public Multimap<Attribute, AttributeModifier> getAttributeModifiers(SlotContext slotContext, UUID uuid, ItemStack stack)
```
do the attribute modifiers reset properly when the plate runs out of charge? or does it stick around till you equip and unequip it

leafreynolds reviewed (changes requested):
Apparently I never pushed submit on my review 🙃

leafreynolds commented:
For good measure, I wanna see what happens if I request additional review from @claude . Maybe it will catch something I haven't.

leafreynolds commented:
Nothing happens 🙄 I'll figure it out later. Claude's throwing some error for whatever reason.

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/eventHandlers/AllomancyEntityEventHandler.java:
```diff
@@ -110,5 +124,54 @@ public static void onLivingHurtEvent(LivingHurtEvent event)
+						// Reduce damage taken by entity
+						event.setAmount(remaining);
+
+						stack.setDamageValue((int) (stack.getDamageValue() - absorbed));
```
Pretty sure damage value isn't 'item health', it's how much damage it's taken. 0 = full/max=broken. so subtracting would make it 'heal'.

You  might need to test this one in game, break point it maybe and see what it's doing.

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/eventHandlers/AllomancyEntityEventHandler.java:
```diff
@@ -110,5 +124,54 @@ public static void onLivingHurtEvent(LivingHurtEvent event)
+		{
+			if(source.is(type))
+			{
+				Optional<ICuriosItemHandler> sub = CuriosApi.getCuriosInventory(entity).resolve();
```
Is this var used for anything ? otherwise it's a slightly more expensive call

leafreynolds commented on src/api/java/leaf/cosmere/api/helpers/CuriosHelper.java:
```diff
@@ -0,0 +1,132 @@
+
+	public static SlotResult getSlot(LivingEntity entity, String slotID, int index) {
+		ICuriosItemHandler sub = getCuriosHandler(entity);
+		assert sub != null;
```
Assertions are disabled at runtime by default in Java, so should probs replace this with a null check if that's what you were wanting

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/eventHandlers/AllomancyEntityEventHandler.java:
```diff
@@ -110,5 +124,54 @@ public static void onLivingHurtEvent(LivingHurtEvent event)
+					{
+						ItemStack stack = slotResult.stack();
+						MistcloakItem item = (MistcloakItem) stack.getItem();
+						if (item.getDamage(stack) == 0)
```
as with other comment, 0 would be max durability I think? needs checking

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/eventHandlers/AllomancyEntityEventHandler.java:
```diff
@@ -110,5 +124,54 @@ public static void onLivingHurtEvent(LivingHurtEvent event)
+				if (CuriosHelper.getCuriosHandler(entity) != null)
+				{
+
+					for (SlotResult slotResult : CuriosHelper.getSlotsWithItem(entity, AllomancyItems.MISTCLOAK.asItem()))
```
we should probably guard against multiple mistcloaks being equippable somehow. Or maybe break out of the loop after reducing the first bit of damage, so people can't double up.

leafreynolds commented on .DS_Store:
Can you delete all these ds store files? If you merge in develop, the should stop trying to add themselves in future by accident.

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/items/MistcloakItem.java:
```diff
@@ -29,7 +40,27 @@ public final String getArmorTexture(ItemStack stack, Entity entity, EquipmentSlo
+		Multimap<Attribute, AttributeModifier> defaultModifiers;
+
+		builder.putAll(ICurioItem.super.getAttributeModifiers(slotContext, uuid, stack));
+			builder.put(Attributes.ARMOR, new AttributeModifier(uuid, "Mistcloack modifier", 2, AttributeModifier.Operation.ADDITION));
```
typo in mistcloak, though it doesn't super matter for this field

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/items/MistcloakItem.java:
```diff
@@ -29,7 +40,27 @@ public final String getArmorTexture(ItemStack stack, Entity entity, EquipmentSlo
+		builder.putAll(ICurioItem.super.getAttributeModifiers(slotContext, uuid, stack));
+			builder.put(Attributes.ARMOR, new AttributeModifier(uuid, "Mistcloack modifier", 2, AttributeModifier.Operation.ADDITION));
+			builder.put(ForgeMod.ENTITY_GRAVITY.get(), new AttributeModifier(uuid, "Mistcloak glide", -0.02, AttributeModifier.Operation.ADDITION));
+			builder.put(Attributes.FLYING_SPEED, new AttributeModifier(uuid, "Mistcloack fly", 1.02, AttributeModifier.Operation.MULTIPLY_BASE));
```
typo in mistcloak, though it doesn't super matter for this field

leafreynolds commented on src/api/java/leaf/cosmere/api/helpers/CuriosHelper.java:
```diff
@@ -0,0 +1,132 @@
+		}
+		return sub.findCurios(item);
+	}
+	public static List<SlotResult> getSlotsWithItem(LivingEntity entity, Item... items)
```
You call getSlotsWithItem 3 times here in a single function, and it does it for every item you're looping through.

call it once and assign it to a variable

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/eventHandlers/AllomancyEntityEventHandler.java:
```diff
@@ -110,5 +124,54 @@ public static void onLivingHurtEvent(LivingHurtEvent event)
+
 	}
+
+	private static ArrayList<ResourceKey<DamageType>> protectedDamageTypes = new ArrayList<>(
```
Can be simplified to something like this, I think? 
```
private static final List<ResourceKey<DamageType>> protectedDamageTypes = 
List.of(
  DamageTypes.FALL,
  DamageTypes.FLY_INTO_WALL,
  DamageTypes.FREEZE
);
```

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/client/render/renderer/MistcloakRenderer.java:
```diff
@@ -52,9 +52,12 @@ public <T extends LivingEntity, M extends EntityModel<T>> void render(
 		this.model.prepareMobModel(entity, limbSwing, limbSwingAmount, partialTicks);
 		this.model.setupAnim(entity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
 
+
```
A few extra lines added for no reason, lets remove these 🙂

leafreynolds commented on src/datagen/allomancy/java/leaf/cosmere/allomancy/AllomancyCuriosProvider.java:
```diff
@@ -0,0 +1,40 @@
+	{
+		//https://docs.illusivesoulworks.com/curios/slots/data-generation
+
+		this.createSlot("head")
```
I think curios already generates a headslot by default, so we probably don't need to add this ourselves. needs double checking.

SoaringEaqle replied:
Not by default. The slot exists in the game, but we need to assign the player to have one in order for them to use it.

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/eventHandlers/AllomancyEntityEventHandler.java:
```diff
@@ -110,5 +124,54 @@ public static void onLivingHurtEvent(LivingHurtEvent event)
+	private static ArrayList<ResourceKey<DamageType>> protectedDamageTypes = new ArrayList<>(
+			List.of(DamageTypes.FALL,
+					DamageTypes.FLY_INTO_WALL,
+					DamageTypes.FREEZE
```
should we be absorbing freeze damage?

SoaringEaqle replied:
I was thinking that if it's a wool cloak, it'll do something against the cold. I can remove it.

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/items/MistcloakItem.java:
does this still get called, now that this doesn't extend from `ArmorItem` ?

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/items/MistcloakItem.java:
```diff
@@ -29,7 +40,27 @@ public final String getArmorTexture(ItemStack stack, Entity entity, EquipmentSlo
+	{
+		ImmutableMultimap.Builder<Attribute, AttributeModifier> builder = ImmutableMultimap.builder();
+
+		Multimap<Attribute, AttributeModifier> defaultModifiers;
```
this var isn't needed, we can return builder.build() directly at the bottom

leafreynolds commented on src/api/java/leaf/cosmere/api/helpers/CuriosHelper.java:
```diff
@@ -0,0 +1,132 @@
+	{
+		if(getSlot(entity, slotID, index) == null)
+		{
+			return null;
```
should we be returning null here? usually we return an optional

leafreynolds commented on src/api/java/leaf/cosmere/api/helpers/CuriosHelper.java:
```diff
@@ -0,0 +1,132 @@
+	{
+		if(getSlot(entity, slotID, index) == null)
+		{
+			return null;
```
should we be returning null here? usually we return an optional

SoaringEaqle committed:
Merge branch 'leafreynolds:develop' into mistcloak_V2

SoaringEaqle committed:
Satisfies requested changes

SoaringEaqle commented:
done

leafreynolds commented on src/surgebinding/java/leaf/cosmere/surgebinding/common/items/ShardbladeItem.java:
```diff
@@ -1,3 +1,4 @@
+
```
Still have a few files in the change list with random lines added, but I could probably go through and remove these? otherwise, the changes are looking good :)

leafreynolds commented:
@claude review

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/client/render/model/MistcloakModel.java:
```diff
@@ -12,6 +13,7 @@
 import com.mojang.blaze3d.vertex.PoseStack;
 import com.mojang.blaze3d.vertex.VertexConsumer;
 import leaf.cosmere.allomancy.common.Allomancy;
+import net.minecraft.client.model.HumanoidArmorModel;
```
`HumanoidArmorModel` is imported but never used — can be removed.

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/items/MistcloakItem.java:
```diff
@@ -1,35 +1,57 @@
+import net.minecraft.world.entity.ai.attributes.Attribute;
+import net.minecraft.world.entity.ai.attributes.AttributeModifier;
+import net.minecraft.world.entity.ai.attributes.Attributes;
 import net.minecraft.world.item.ArmorItem;
```
Now that `MistcloakItem` extends `Item` instead of `ArmorItem`, these imports are unused and should be cleaned up:
- `ArmorItem` (line 16)
- `ArmorMaterial` (line 17)
- `Entity` (line 11)
- `EquipmentSlot` (line 12)
- `Nonnull` (line 24)

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/items/MistcloakItem.java:
```diff
@@ -1,35 +1,57 @@
+		builder.putAll(ICurioItem.super.getAttributeModifiers(slotContext, uuid, stack));
+			builder.put(Attributes.ARMOR, new AttributeModifier(uuid, "Mistcloak modifier", 2, AttributeModifier.Operation.ADDITION));
+			builder.put(ForgeMod.ENTITY_GRAVITY.get(), new AttributeModifier(uuid, "Mistcloak glide", -0.02, AttributeModifier.Operation.ADDITION));
+			builder.put(Attributes.FLYING_SPEED, new AttributeModifier(uuid, "Mistcloak fly", 1.02, AttributeModifier.Operation.MULTIPLY_BASE));
```
`MULTIPLY_BASE` with value `1.02` means the base flying speed gets an additional `1.02 * base` added, resulting in **+102%** flying speed — not +2%.

If a small boost is intended, use `0.02` instead. Or if the intent is to roughly double flying speed, then `1.0` with `MULTIPLY_BASE` would be clearer.

Also, the `builder.put` lines on 48-50 have an extra level of indentation compared to `builder.putAll` on line 47 and `return` on line 51 — looks like a formatting accident.

Gerbagel replied:
Is this even true? Claude doesn't cite any documentation of anything. Probably something to check, at least. Worst case it just hallucinated, as usual.

SoaringEaqle replied:
It is seemingly right about this.

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/eventHandlers/AllomancyEntityEventHandler.java:
```diff
@@ -110,5 +124,54 @@ public static void onLivingHurtEvent(LivingHurtEvent event)
+						// Reduce damage taken by entity
+						event.setAmount(remaining);
+
+						stack.setDamageValue((int) (stack.getDamageValue() + absorbed));
```
Two issues here:

1. **No overflow protection** — if the item only has 5 durability left but `absorbed` is 10, `damageValue` will exceed `maxDamage`. Should clamp or break the item:
```java
int newDamage = Math.min(stack.getDamageValue() + Mth.ceil(absorbed), stack.getMaxDamage());
stack.setDamageValue(newDamage);
```

2. **Fractional truncation** — for small hits (e.g. 3 HP), `absorbed = 0.6f`, which truncates to `0` durability cost via the `(int)` cast. The item absorbs damage for free. Consider using `Mth.ceil()` instead.

Gerbagel replied:
The item breaking might be handled somewhere else. Again, something to check.

SoaringEaqle replied:
I went ahead and forced it to happen any way, plus did fix the math up a bit.

leafreynolds commented on src/allomancy/java/leaf/cosmere/allomancy/common/eventHandlers/AllomancyEntityEventHandler.java:
```diff
@@ -110,5 +124,54 @@ public static void onLivingHurtEvent(LivingHurtEvent event)
+		{
+			if(source.is(type))
+			{
+				if (CuriosHelper.getCuriosHandler(entity) != null)
```
`getCuriosHandler()` returns `Optional<ICuriosItemHandler>`, not a nullable reference. This `!= null` check will always be true (an Optional is never null). Should be:
```java
if (CuriosHelper.getCuriosHandler(entity).isPresent())
```
Or better yet, skip this check entirely since `getSlotsWithItem` on the next line will just return an empty list if there's no handler.

Gerbagel replied:
Already handled.

leafreynolds commented on src/api/java/leaf/cosmere/api/helpers/CuriosHelper.java:
```diff
@@ -0,0 +1,138 @@
+
+import java.util.*;
+
+public class CuriosHelper
```
Per project conventions, the `api` source set should remain dependency-free from external module implementations. This class depends directly on `CuriosApi` / `ICuriosItemHandler`. Consider moving it to `src/main/java/leaf/cosmere/common/helpers/` instead.

leafreynolds replied:
ignore

leafreynolds commented on src/api/java/leaf/cosmere/api/CosmereTags.java:
```diff
@@ -37,6 +38,7 @@ public static class Items
 		public static final TagKey<Item> CURIO_RING = makeItem(CuriosApi.MODID, "ring");
 		public static final TagKey<Item> CURIO_LEGS = makeItem(CuriosApi.MODID, "legs");
 		public static final TagKey<Item> CURIO_FEET = makeItem(CuriosApi.MODID, "feet");
+		public static final TagKey<Item> CURIO_SHARDPLATE = makeItem(CuriosApi.MODID, "shardplate");
```
`CURIO_SHARDPLATE` appears to be part of a separate shardplate-as-curio feature. Consider splitting this into its own PR to keep this one focused on the mistcloak.

Gerbagel replied:
Ignore. This is true to keep in mind for future PRs, but it's really not so important.

leafreynolds reviewed (commented):
## Mistcloak v2 Review

Good concept converting the Mistcloak from ArmorItem to a Curios ICurioItem. The model rework is a major simplification — dropping all those complex rotated sub-parts in favor of a cleaner root-based hierarchy is a big readability win.

### Bugs
- **Flying speed modifier** uses `MULTIPLY_BASE` with value `1.02`, which gives **+102% base flying speed**, not +2%. If a small boost was intended, the value should be `0.02`.
- **Durability overflow** — absorbed damage can push `damageValue` past `maxDamage` without the item breaking. Need a bounds check.
- Damage value semantics were already flagged by @leafreynolds — `damageValue` is damage *taken* (0 = pristine), so the guard and arithmetic need adjusting.

### Cleanup Needed
- Several now-unused imports after the `ArmorItem` → `Item` conversion (`ArmorItem`, `ArmorMaterial`, `Entity`, `EquipmentSlot`, `Nonnull`)
- `CuriosHelper` is placed in the `api` source set but depends on CuriosApi — per project conventions, `api` should remain dependency-free from external module implementations
- `CURIO_SHARDPLATE` tag addition and multiple surgebinding whitespace-only changes seem unrelated to this PR's scope
- Unused `ForgeMod` import in `AllomancyRecipeGen`

### Missing
- I don't see a curio slot tag entry (e.g. `data/curios/tags/items/head.json`) that registers the mistcloak item for the head slot. Without this, Curios won't know the item can be equipped there. Worth verifying this works in-game.

See inline comments for specifics.

leafreynolds commented on src/datagen/allomancy/java/leaf/cosmere/allomancy/AllomancyRecipeGen.java:
```diff
@@ -12,6 +12,7 @@
 import net.minecraft.data.recipes.*;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.item.Items;
+import net.minecraftforge.common.ForgeMod;
```
Unused import — `ForgeMod` isn't referenced anywhere in this file.

Gerbagel replied:
I'm not gonna check every unused import. If they're removed, cool. If not, no huge loss, someone'll probably catch them later.

SoaringEaqle committed:
Code Cleanup

Removes unused imports.
Re-does AllomancyEntityEventHandler.java to check that an entity has a mistcloak before checking if the damage type can be reduced by one.
Fixes math in AllomancyEntityEventHandler.java, to make sure it doesn't go below max durability. Breaks a mistcloak when it reaches 0 durability.

SoaringEaqle committed:
Merge branch 'leafreynolds:develop' into mistcloak_V2

leafreynolds committed:
Remove .DS_Store and revert unrelated surgebinding whitespace changes
